packages/core/test/transient.test.ts

```diff
@@ -670,4 +676,51 @@ describe('transient states (eventless transitions)', () => {
     const service = interpret(machine);
     expect(() => service.start()).not.toThrow();
   });
+
+  it('should be taken even in absence of other transitions', (done) => {
+    const machine = createMachine<{
+      ref?: ActorRef<any>;
+    }>({
+      initial: 'a',
+      context: {
+        ref: undefined
+      },
+      states: {
+        a: {
+          entry: assign<any>({
+            ref: () =>
+              spawn(
+                createMachine({
+                  initial: 'waiting',
+                  states: {
+                    waiting: { after: { 100: 'done' } },
+                    // This will send the parent a "done.invoke.*" event,
+                    // which is not handled by the parent.
+                    done: { type: 'final' }
+                  }
+                })
+              )
+          }),
+          // There will be no enabled transitions for the "done.invoke.*" event
+          // but there will be transient transitions.
+          always: {
+            target: 'b',
+            cond: (ctx: any, _e) => {
+              // This will be reached when the ref is done and emits a "done.invoke.*"
+              // event to the parent
+              return ctx.ref.getSnapshot().matches('done');
+            }
+          }
+        },
+        b: {
+          type: 'final'
+        }
+      }
+    });
+    const service = interpret(machine).onDone(() => {
+      done();
+    });
+
+    service.start();
+  });
```

Comment:
It feels like this test is longer than it could be and as such requires more mental overhead to understand it, whereas the basic requirement here can be represented in a quite simple way:

```suggestion
  it('should be taken even in absence of other transitions', () => {
    let shouldMatch = false;

    const machine = createMachine({
      initial: 'a',
      states: {
        a: {
          always: {
            target: 'b',
            // TODO: in v5 remove `shouldMatch` and replace this guard with:
            // cond: (ctx, ev) => ev.type === 'WHATEVER'
            cond: () => shouldMatch
          }
        },
        b: {}
      }
    });
    const service = interpret(machine).start();

    shouldMatch = true;
    service.send({ type: 'WHATEVER' });

    expect(service.state.value).toBe('b');
  });
```